src/clients/Deluge.ts

```diff
@@ -262,44 +284,34 @@ export default class Deluge implements TorrentClient {
 				});
 				return InjectionResult.FAILURE;
 			}
-
 			const params = this.formatData(
 				`${newTorrent.getFileSystemSafeName()}.cross-seed.torrent`,
 				newTorrent.encode().toString("base64"),
 				path ? path : torrentInfo!.save_path,
-				!!searchee.infoHash,
 				decision,
 			);
 			const addResult = await this.call<string>(
 				"core.add_torrent_file",
 				params,
 			);
-			if (addResult.result) {
-				const { linkingCategory } = getRuntimeConfig();
+
+			const addResponse =
+				typeof addResult.result === "string"
+					? addResult.result
+					: addResult.error;
```

Comment:
imo this `addResponse` variable that represents multiple values just obscures things. The extraction into this.calculateLabel() is good though

Reply:
It might be better to review qbit-tags or we merge now - this is outdated severely.

This one was mainly supposed to be the one we rebase and merge into master, so to keep the commits clean enough that that is possible, we branched - qbit has been a mess and therefore we've iterated through probably double digit attempts to nail down this category/tagging/autotmm - that's in qbit-tags. id-search is rebased off qbit-tags presently, and ensemble is rebased off id-search.

gitception.

Reply:
> imo this `addResponse` variable that represents multiple values just obscures things. The extraction into this.calculateLabel() is good though

Not sure how to do this though, given the different properties and their types..

Reply:
It seems logical to do the logic off of one variable that we can either be reliant on success based on its type/properties being a string, or of ErrorType in case it fails.

We couple the 4 states this way, it allows the LSP to show what state was eliminated after each check